client/engine/engine.go

```diff
@@ -589,11 +649,18 @@ func (e *Engine) constructObjectiveFromMessage(id protocols.ObjectiveId, ss stat
 		}
 		return &vfo, nil
 	case virtualdefund.IsVirtualDefundObjective(id):
-		vdfo, err := virtualdefund.ConstructObjectiveFromState(ss.State(), false, *e.store.GetAddress(), e.store.GetChannelById, e.store.GetConsensusChannel)
+		voucher, err := e.vm.Voucher(ss.ChannelId())
 		if err != nil {
 			return &virtualfund.Objective{}, fmt.Errorf("could not create virtual fund objective from message: %w", err)
 		}
+
+		vdfo, err := virtualdefund.ConstructObjectiveFromVoucher(ss.State().FixedPart(), voucher, false, *e.store.GetAddress(), e.store.GetChannelById, e.store.GetConsensusChannel)
```

Comment:
Do we do any verification on the voucher? What if Alice sends a maliciously low voucher?